File: infrastructure/apps/api/src/database/seeds/seed.ts
import { Store } from '../entities/store.entity'
import { AppDataSource } from '../../data-source'

const stores = [
  {
    name: 'Extra Supermercado',
    baseUrl: 'https://www.extra.com.br/',
    zipcode: 1000000,
  },
  {
    name: 'Atacadão',
    baseUrl: 'https://www.atacadao.com.br/',
    zipcode: 1000000,
  },
  {
    name: 'mercado carrefour',
    baseUrl: 'https://mercado.carrefour.com.br',
    zipcode: 1000000,
  },
];

const old_stores = [
  {
    name: 'Extra Supermercado',
    sourceUrl: 'https://www.extra.com.br/',
    zipcode: 1000000,
  },
  {
    name: 'Atacadão',
    sourceUrl: 'https://www.atacadao.com.br/',
    zipcode: 1000000,
  },
  {
    name: 'mercado carrefour',
    sourceUrl: 'mercado.carrefour.com.br',
    zipcode: 1000000,
  },
  {
    name: 'Muffato (Super Muffato)',
    sourceUrl: 'https://www.supermuffato.com.br/',
    zipcode: 80000000,
    // active: 0,
  },
  {
    name: 'iFood',
    sourceUrl: 'https://www.ifood.com.br/',
    zipcode: 1000000,
    // active: 0,
  },
  {
    name: 'Rappi',
    sourceUrl: 'https://www.rappi.com.br/',
    zipcode: 1000000,
    // active: 0,
  },
  {
    name: 'Americanas',
    sourceUrl: 'https://www.americanas.com.br/',
    zipcode: 1000000,
    // active: 0,
  },
]

async function seed () {
  const dataSource = await AppDataSource.initialize()
  const repo = dataSource.getRepository(Store)

  console.log('Seeding database...')
  for (const s of stores) {
    const exists = await repo.findOne({ where: { name: s.name } })
    if (!exists) await repo.save(repo.create(s))
  }
  console.log('Done seeding.')
  await dataSource.destroy()
}

seed().catch(console.error)

File: infrastructure/apps/scraper/seed.py
from db import get_conn

async def seed_stores():
    stores = [
        {
            "name": "Extra Supermercado",
            "domain": "extra.com.br",
            "baseUrl": "https://www.extra.com.br/",
            "countryCode": "BR",
            "channel": "online",
        },
        {
            "name": "Atacadão",
            "domain": "atacadao.com.br",
            "baseUrl": "https://www.atacadao.com.br/",
            "countryCode": "BR",
            "channel": "online",
        },
        {
            "name": "Mercado Carrefour",
            "domain": "mercado.carrefour.com.br",
            "baseUrl": "https://mercado.carrefour.com.br/",
            "countryCode": "BR",
            "channel": "online",
        },
    ]

    conn = await get_conn()

    for store in stores:
        existing = await conn.fetchval(
            'SELECT id FROM store WHERE LOWER(name) = LOWER($1)', store["name"]
        )
        if not existing:
            await conn.execute(
                '''
                INSERT INTO store (name, domain, "baseUrl", "countryCode", channel)
                VALUES ($1, $2, $3, $4, $5)
                ''',
                store["name"],
                store["domain"],
                store["baseUrl"],
                store["countryCode"],
                store["channel"],
            )
            print(f"✅ Seeded store: {store['name']}")
        else:
            print(f"ℹ️ Store already exists: {store['name']}")

    await conn.close()
